Dockerfile.ubi.ci

```
@@ -0,0 +1,211 @@
+FROM base AS bnxt
+RUN PYTHONUNBUFFERED=1 dnf install -y \
+    autoconf automake \
+    gcc gcc-c++ make \
+    iproute kmod \
+    libibverbs-devel librdmacm-devel librdmacm \
+    libtool pciutils \
+    python systemd wget && \
+    dnf clean all && \
+    rm -rf /var/cache/dnf
+
+RUN wget "https://docs.broadcom.com/docs-and-downloads/ethernet-network-adapters/NXE/BRCM_${BCM_DRIVER_VERSION}/bcm_${BCM_DRIVER_VERSION}.tar.gz" && \
+    tar -xvzf "bcm_${BCM_DRIVER_VERSION}.tar.gz" && \
+    tar -C /tmp -xf "${WORKDIR}/bcm_${BCM_DRIVER_VERSION}/drivers_linux/bnxt_rocelib/libbnxt_re-${BCM_LIBBNXT_RE_VERSION}.tar.gz" && \
+    cd "/tmp/libbnxt_re-${BCM_LIBBNXT_RE_VERSION}" && \
+    sh autogen.sh && \
+    ./configure && \
+    make clean && \
+    make && \
+    make install
+  
+################################################################################
+FROM base AS driver
+
+RUN PYTHONUNBUFFERED=1 dnf install -y wget rpm && \
+    dnf clean all
+
+# Add Oracle Linux RDMA repo to install rdma-core and libibverbs-utils
+RUN cat <<'EOF' > /etc/yum.repos.d/oracle_rdma.repo
+[ol9-rdma]
+name=Oracle Linux $releasever RDMA Packages
+baseurl=http://yum.oracle.com/repo/OracleLinux/OL9/RDMA/x86_64/
+enabled=1
+gpgcheck=0
+EOF
+
+RUN PYTHONUNBUFFERED=1 dnf install -y dnf-plugins-core && \
+    dnf clean all && \
+    dnf makecache && \
+    dnf install -y \
+        bzip2-libs \
+        libdrm \
+        libibverbs \
+        librdmacm \
+        rdma-core \
+        libibverbs-utils \
+        kmod \
+        pciutils && \
+    dnf clean all && \
+    rm -rf /var/cache/dnf && \
+    ln -sf /usr/lib64/libbz2.so.1 /usr/lib64/libbz2.so.1.0 && \
+    ldconfig
+
+COPY --from=bnxt /usr/local/lib/libbnxt_re-rdmav34.so /usr/local/lib/libbnxt_re-rdmav34.so
+RUN if [ -f /usr/lib64/libibverbs/libbnxt_re-rdmav34.so ]; then \
+        mv /usr/lib64/libibverbs/libbnxt_re-rdmav34.so /usr/lib64/libibverbs/libbnxt_re-rdmav34.so.inbox; \
+    fi && \
+    ln -s /usr/local/lib/libbnxt_re-rdmav34.so /usr/local/lib/libbnxt_re.so && \
+    ldconfig
+
+COPY --from=atom /atom/etc/ /etc/
+COPY --from=atom /atom/opt/ /opt/
+COPY --from=atom /atom/usr/ /usr/
+COPY --from=atom /atom/lib/ /usr/
+
+################################################################################
+FROM driver AS python
+
+RUN PYTHONUNBUFFERED=1 dnf install -y \
+    gcc \
+    gcc-c++ \
+    make \
+    openssl-devel \
+    bzip2-devel \
+    libffi-devel \
+    zlib-devel \
+    xz-devel \
+    sqlite-devel \
+    wget \
+    tar && \
+    dnf clean all
+
+ARG PYTHON_VERSION
+RUN cd /tmp && \
+    wget "https://www.python.org/ftp/python/${PYTHON_VERSION}/Python-${PYTHON_VERSION}.tgz" && \
+    tar -xzf "Python-${PYTHON_VERSION}.tgz" && \
+    cd "Python-${PYTHON_VERSION}" && \
+    ./configure --enable-optimizations --with-ensurepip=install && \
+    make && \
+    make altinstall && \
+    cd / && \
+    rm -rf "/tmp/Python-${PYTHON_VERSION}"*
```

Would it be possible to bump python to 3.12 (or 3.11)? Those come packaged in the UBI image, so you'd just be able to

```suggestion
RUN dnf install python${PYTHON_VERSION}
```

---

I got confused - I didn't install python3.12 with dnf install, I actually installed python3.10. At that time, dnf installation didn't work, so I downloaded Python directly.
Can you tell me how to install python3.10 using dnf install?

---

Ah ha, I see.
Since the `Dockerfile.ubi.ci` file is a Dockerfile for supporting UBI, we don't need to consider Python 3.10, which cannot be installed via dnf.
It seems we can just support Python `3.9`, `3.11`, and `3.12`, which are installable via dnf.
Thank you for sharing the information.